include/merlin/external_storage.cuh

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+#include "merlin/memory_pool.cuh"
+
+namespace nv {
+namespace merlin {
+
+template <class Key, class Value>
+class ExternalStorage {
```

Comment:
There are two problems in the initial design: the 1st one is the evict mechanism including meta procession hasn't been considered; the 2nd one is the configuration items of the ExtendStorage should be a unified struct not the separate input of the API, it's hard to expand new configuration items in the future.

Reply:
1) Meta data will get lost, as external storage proposed by GDS cannot store metadata.
2) I can understand why you would want an option struct. But I don't really see why you want to entangle the external storage and the hash table. IMHO the hashtable best shouldn't know any details about the external storage.

Reply:
Scope of PR changed.